WISE-Community/WISE
# Show posts from Discourse in teacher homepage

Note: needs styling from @breity

This PR will show up to 3 most recent topics from a linked Discourse site for the logged-in teacher.

### Set up: Discourse
1. Install Discourse with Docker by following instructions on this page: https://meta.discourse.org/t/beginners-guide-to-install-discourse-for-development-using-docker/102009. This will create the admin user.
2. Enable Cross-Origin Resource Sharing (CORS) for Discourse so that WISE can request for latest topics. In discourse/bin/docker/unicorn, add ```-e DISCOURSE_ENABLE_CORS='true'``` to the docker command.
```
...
docker exec -it -e DISCOURSE_ENABLE_CORS='true' -u discourse:discourse discourse_dev /bin/bash -c "$CMD"
```
3. Start Discourse by running ```d/unicorn -x``` and check that it's running at http://localhost:9292
4. Log in with the admin user that you created as part of step 1, and configure CORS and SSO settings

Specify CORS origin:
<img width="842" alt="Screen Shot 2020-11-13 at 9 38 32 AM" src="https://user-images.githubusercontent.com/119416/99103391-58cab800-2594-11eb-8ac7-4e37f2b706d0.png">

Enable SSO:
<img width="566" alt="Screen Shot 2020-11-13 at 9 39 00 AM" src="https://user-images.githubusercontent.com/119416/99103428-64b67a00-2594-11eb-880e-df9b7695e889.png">

Specify SSO url and secret (the secret must match the setting in application.properties, which we'll get to later):
<img width="488" alt="Screen Shot 2020-11-13 at 9 39 36 AM" src="https://user-images.githubusercontent.com/119416/99103436-67b16a80-2594-11eb-9b0b-a53880f82e72.png">



### Set up: WISE
1. Link to the Discourse instance by setting properties in application.properties
```
discourse_url=http://localhost:9292
discourse_sso_secret_key=secret_handshake
```

### Test
1. Log in as a teacher user. In the teacher homepage, you should see a "Recent Posts in Teacher Community" section at the top of the page, listing up to three latest topics. There should also be an expand icon.
2. Click on the expand icon. This should open a new tab, and log you into Discourse using your WISE account.
3. Add a new topic in Discourse, and verify that the Recent Posts section updates and show the new topic at the top.
4. Log in as another teacher user. In the teacher homepage, click on a topic. It should take you to the topic page in Discourse (but not log you in yet). If you click on the 'reply' button, it should log you into Discourse using your WISE account.
5. Comment out the properties from application.properties, and restart. The "Recent Posts in Teacher Community" section should not appear.

Closes #2806
